packages/block-library/src/template-part/edit/utils/search.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Filters an item list given a search term.
+ *
+ * @param {Array}  items       Item list
+ * @param {string} searchValue Search input.
+ *
+ * @return {Array} Filtered item list.
+ */
+export function searchItems( items, searchValue ) {
+	if ( ! searchValue ) {
+		return items;
+	}
+
+	const normalizedSearchValue = searchValue.toLowerCase();
+	return items.filter( ( item ) => {
+		const normalizedTitle = item.title.toLowerCase();
+
+		return normalizedTitle.includes( normalizedSearchValue );
+	} );
```

Comment:
This is a very basic search. I'm not sure if we need a ranked search like inserter here.

Reply:
I think it's ok to start with something simple, but also looks like it probably wouldn't be too difficult to improve it. I had a look at the code for the inserter to see how it works: https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inserter/search-items.js#L145-L161

The biggest win would probably be to still consider it a match if the search term words are in the title but in a different order (e.g. 'image light' and 'light image' produce the same search results), which seems to be what this bit of code does using the `words` function:
https://github.com/WordPress/gutenberg/blob/dfc4b8b8b831c6d5e2c280b06c5871d91e66f337/packages/block-editor/src/components/inserter/search-items.js#L144-L162

It'll be a bit easier for patterns since there's only one `title` field to check against.

Reply:
>This is a very basic search. I'm not sure if we need a ranked search like inserter here.

We don't actually have a `ranking` system right now and the one we need would be different than the inserter, that uses `frecency`.

Having said that, I think we can start small with a couple of tweaks maybe:
1. Rename the function to something more specific(ex. `searchPatterns`) as is not generic as some other similar functions
2. Probably add simple `accents, trim` handling to `normalizedTitle`.

In general I think integrating `pattern explorer` here would be fitting, but until we have something like that, it's okay to have at least a basic search by title. Also when we'll work on a patterns `ranking` system we might convert similar functions to a `selector`..

Reply:
Addressed in d16c616c1b25f9a9d2f02bab03eacecc672ded82.